src/facade/bank/BankFacade.ts

```diff
@@ -0,0 +1,311 @@
+import { Request, Response } from "express";
+import Config from "src/config/Config";
+import CardDto from "src/dto/bank/CardDto";
+import CustomError from "src/error/CustomError";
+import AuctionService from "src/service/auction/AuctionService";
+import AuctionPhotoService from "src/service/auctionPhoto/AuctionPhotoService";
+import BankService from "src/service/bank/BankService";
+import PhotoService from "src/service/photo/photoService";
+import PhotographerPaymentService from "src/service/photographerPayment/PhotographerPaymentService";
+import PurchasedPhotoService from "src/service/purchasedPhoto/PurchasedPhotoService";
+import StatusService from "src/service/status/StatusService";
+import UserService from "src/service/user/userService";
+import VoteService from "src/service/vote/VoteService";
+import BankValidation from "src/validation/bank/BankValidation";
+import BaseValidation from "src/validation/base/BaseValidation";
+
+class BankFacade {
+	private bankService: BankService;
+	private bankValidation: BankValidation;
+	private userService: UserService;
+	private statusService: StatusService;
+	private baseValidation: BaseValidation;
+	private photoService: PhotoService;
+	private auctionPhotoService: AuctionPhotoService;
+	private purchasedPhotoService: PurchasedPhotoService;
+	private voteService: VoteService;
+	private photographerPaymentService: PhotographerPaymentService;
+	private auctionService: AuctionService;
+
+	constructor() {
+		this.bankService = new BankService();
+		this.bankValidation = new BankValidation();
+		this.userService = new UserService();
+		this.statusService = new StatusService();
+		this.baseValidation = new BaseValidation();
+		this.photoService = new PhotoService();
+		this.auctionPhotoService = new AuctionPhotoService();
+		this.purchasedPhotoService = new PurchasedPhotoService();
+		this.voteService = new VoteService();
+		this.photographerPaymentService = new PhotographerPaymentService();
+		this.auctionService = new AuctionService();
+	}
+
+	public async approveUser(req: Request, res: Response): Promise<void> {
+		let cardDto: CardDto;
+		let banUser: boolean = false;
+
+		// get valid body from request
+		try {
+			cardDto = await this.bankValidation.validateBankCredentials(req);
+		} catch (error: any) {
+			CustomError.handleError(res, error);
+			return;
+		}
+
+		// add provision
+		try {
+			await this.bankService.addProvision({ ...cardDto, provision: Config.provisionAmount });
+		} catch (error: any) {
+			if (error.message === "Cannot perform bank api operation. Balance is not sufficient for provision.") banUser = true;
```

Comment:
banlamayı sadece provizyon geçerli olup eğer kişi auctionu kazanırsa son yaptığı bid'in tüm değerini çekerken hata olursa yapmayacak mıydık? şu anki durumda sadece yetersiz bakiye diye hata vermesi daha mantıklı olur bence